### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by thatbudakguy**
refs #103 

## dev note
tl;dr important stuff is the `Pagination` component and `pages.js` module. it ended up being easiest to just replicate the logic from the django pagination template in js after all; upside is that we can use this pagination component on any project to get the same pagination.

I started to break out pieces of the vuex store into [modules](https://vuex.vuejs.org/guide/modules.html), which was super useful. The methods have the same API, they're just part of more self-contained units now and are namespaced, so you can do stuff like `dispatch('results/reload')` or `getters.results.total`.

I played around a little bit with modularizing the facet-related state in `modules/facets.js` but left this for later since it wasn't necessary for the pagination to work. The file isn't used by vuex currently.

### CODE REVIEW

**Comment by thatbudakguy** [comment]
Kind of a bummer that I didn't have unit tests to catch these regressions, but I'm glad you were able to.
Updated the following cases to cause a reset to page 1:

- changing range facet min/max
- changing keyword query
- toggling a facet choice
- clearing all active choices

Also fixed the total result count display and URL updating. It seems like there's a case where the initial load of results will fail, but I'm having trouble isolating what it is...hopefully will get to write some tests for this soon so I have a little more help keeping it stable.

**Comment by rlskoeser** [comment]
@thatbudakguy looks like that fixed everything I caught before
